ignite/contrib/handlers/time_profilers.py

```diff
@@ -437,3 +438,317 @@ def odict_to_str(d: Mapping):
         )
         print(output_message)
         return output_message
+
+
+class HandlersTimeProfiler:
+    """
+    HandlersTimeProfiler can be used to profile the handlers,
+    data loading and data processing times.
+
+    Examples:
+
+    .. code-block:: python
+
+        from ignite.contrib.handlers import HandlersTimeProfiler
+
+        trainer = Engine(train_updater)
+
+        # Create an object of the profiler and attach an engine to it
+        profiler = HandlersTimeProfiler()
+        profiler.attach(trainer)
+
+        @trainer.on(Events.EPOCH_COMPLETED)
+        def log_intermediate_results():
+            profiler.print_results(profiler.get_results())
+
+        trainer.run(dataloader, max_epochs=3)
+
+        profiler.write_results('path_to_dir/time_profiling.csv')
+
+    """
+
+    EVENT_FILTER_THESHOLD_TIME = 0.0001
+
+    def __init__(self):
+        self._dataflow_timer = Timer()
+        self._processing_timer = Timer()
+        self._event_handlers_timer = Timer()
+
+        self.dataflow_times = None
+        self.processing_times = None
+        self.event_handlers_times = None
+
+    @staticmethod
+    def _get_callable_name(handler: Callable):
+        # get name of the callable handler
+        return getattr(handler, "__qualname__", handler.__class__.__name__)
+
+    def _create_wrapped_handler(self, handler: Callable, event: Events):
+        @functools.wraps(handler)
+        def _timeit_handler(*args: Any, **kwargs: Any):
+            self._event_handlers_timer.reset()
+            handler(*args, **kwargs)
+            t = self._event_handlers_timer.value()
+            hname = self._get_callable_name(handler)
+            # filter profiled time if the handler was attached to event with event filter
+            if not hasattr(handler, "_parent") or t > self.EVENT_FILTER_THESHOLD_TIME:
+                self.event_handlers_times[event][hname].append(t)
+
+        # required to revert back to original handler after profiling
+        setattr(_timeit_handler, "_profiler_original", handler)
+        return _timeit_handler
+
+    def _timeit_processing(self):
+        # handler used for profiling processing times
+        t = self._processing_timer.value()
+        self.processing_times.append(t)
+
+    def _timeit_dataflow(self):
+        # handler used for profiling dataflow times
+        t = self._dataflow_timer.value()
+        self.dataflow_times.append(t)
+
+    def _reset(self, event_handlers_names):
+        # reset the variables used for profiling
+        self.dataflow_times = []
+        self.processing_times = []
+
+        self.event_handlers_times = {e: {h: [] for h in event_handlers_names[e]} for e in event_handlers_names}
+
+    @staticmethod
+    def _is_internal_handler(handler: Callable):
+        # checks whether the handler is internal
+        return "HandlersTimeProfiler." in repr(handler)
```

Comment:
```suggestion
        return any(n in repr(handler) for n in ["HandlersTimeProfiler.", "Timer."])
```
We have to ignore timers too, I think